src/views/product-landing/helpers/make-heading-slug-scope.ts

```diff
@@ -0,0 +1,31 @@
+import slugify from 'slugify'
+
+/**
+ * Returns a function that generates slugs from heading strings,
+ * while ensuring those slugs are unique across calls of the returned function
+ * by suffixing with a `-{number}` where needed.
+ */
+export function makeHeadingSlugScope(): (string) => string {
+  /**
+   * Track which headings slugs have been used, to avoid potential duplicates
+   */
+  const USED_SLUGS = []
+
+  /**
+   * Generate a heading slug, but avoid potential duplicates.
+   * Duplicates are determined based on USED_SLUGS in this scope.
+   */
+  function makeHeadingSlug(heading: string): string {
+    let suffix = 1
+    const baseSlug = slugify(heading, { lower: true })
+    let headingSlug = baseSlug
+    while (USED_SLUGS.indexOf(headingSlug) !== -1) {
```

Comment:
Do you want to loop here? Or just use an `if` statement? 🤔

Reply:
I went for a `while` loop for those ✨ rare ✨ cases where there are many multiples of the same heading (eg [this tutorial](https://learn.hashicorp.com/tutorials/terraform/azure-change?in=terraform/azure-get-started#apply-your-changes-1) has multiple `Apply your changes` headers).

In such cases I _think_ the logic is simpler with `while` - though perhaps not quite as performant, not sure though, seems like performance might actually be better with `while` as we can avoid regex & `parseInt` work that would potentially be required to handle things with `if` rather than looping.

That being said I may not have conceived of the cleanest `if` approach... I might be overthinking it though... Here's my brain dump:

<details>

<summary>Possible `if` implementation</summary>

As a concrete example, consider a case where there are three identical `Apply your changes` headings. This is my thought process through trying to do things without a `while` loop:

1.  for the first heading, the initial `baseSlug` is `apply-your-changes`
    - No problemo, no duplicates yet, the returned `headingSlug` is `apply-your-changes`
    - USED_SLUGS is now `[ "apply-your-changes" ]`
2. for the second heading, the initial `baseSlug` is `apply-your-changes` again
    - We get a conflict
    - In this specific "second duplicate" case we only have one previous duplicate, so we could filter for exact duplicates of the `baseSlug` only, and tack `-1` onto the conflicting heading
    - For the first duplicate, this doesn't feel too complicated, we end up with `apply-your-changes-1`. Yay.
    - USED_SLUGS is now `[ "apply-your-changes", "apply-your-changes-1" ]`
3. for third heading, initial `baseSlug` is `apply-your-changes` once again 
    - We get a conflict, `"apply-your-changes"` is already in `USED_SLUGS`
    - The "filter for exact duplicates and tack something on" strategy falls down here, we'd end up with eg `apply-your-changes-1` again which could be a duplicate (in this case, and any "third duplicate" case, it would be guaranteed to be a duplicate!)
    - We could re-check for duplicates after the tack-on, but then we're back in a `while` loop. To stick with a single `if` we need to rethink, I think.
    - I think we'd want to detect any suffix we've previously added in all `USED_SLUGS` - maybe with a regex like `-\d+$`. 
        - In rare cases this might catch authored headings that happen to have the same suffix - that's fine, in fact good & necessary I think
    - We'd have multiple matches (`[ "apply-your-changes", "apply-your-changes-1" ]`), and we'd have to figure out which has the highest index, I think - otherwise we might generate a suffixed slug that already exists in `USED_SLUGS`.
    - So we'd need to sort these matches for the `${baseSlug}-\d+$` to get the one with the highest suffixed integer, probably using `parseInt` (simple `sort` also seems possible, maybe faster? though might not catch weird edge cases eg if an authored heading slugs out to `something-02` or whatnot)
    - After a bit of filtering, `parseInt`-ing, and then an increment of the highest detected integer suffic, we'd end up with `apply-your-changes-2`
    - I think this approach would work for fourth, fifth, `...n` headings

Definitely feels possible now that I've thought through it, but IMO the relatively minor performance gain (and possibly only more performant in cases where there are _many_ duplicates) is possibly not worth the extra complexity.

</details>